api/interface.go

```diff
@@ -143,10 +149,15 @@ type LoginResultParams struct {
 // EnsureTag should be used when a login provider needs to ensure
 // a login result has a tag set, particularly in cases where the
 // server doesn't return a user identity.
-func (l *LoginResultParams) EnsureTag(tag names.Tag) {
-	if l.tag == nil {
+func (l *LoginResultParams) EnsureTag(tagStr string) error {
+	if l.tag == nil && tagStr != "" {
```

Comment:
I'd like a second opinion on this. I added the extra check here `&&  tagStr != ""` because some tests were failing, e.g. `TestEndpointFiltering` where it seems to setup a fake controller that doesn't respond with a tag. So I'm not sure if this is the right solution.

Reply:
We should pass in a tag here. In hindsight after seeing this we should not have changed the p.tag to p.userInfo. If we need to ultimately pass around a tag, leave the tag on the provider and adjust that logic in there. Round tripping between a tag and string is not what we'd want to do. Then all this logic here can be reverted.

Reply:
Good point, I've done that and added some tests to verify things work if a nil Tag is passed in.